torch/_ops.py

```diff
@@ -449,10 +455,14 @@ def check_overloaded(arg):
         if dispatch_key != DispatchKey.PreDispatch:
             self._dispatch_cache[dispatch_key] = self.py_kernels[final_key]
         kernel = self.py_kernels[final_key]
-        # It's illegal to register DispatchKey to py_kernels, since there's no
-        # C++ kernel to call into
-        assert not isinstance(kernel, DispatchKey)
-        return kernel(*args, **kwargs)
+        if final_key in self.needs_keyset:
+            key_set = _compute_keyset(args, kwargs, self.non_fallthrough_keys)
+            return kernel(key_set, *args, **kwargs)
```

Comment:
I think (have not verified) the problem is the final_key is AutogradCPU but self.needs_keyset contains {Autograd}.

DispatchKey::Autograd is different from DispatchKey::AutogradCPU

Reply:
Design suggestion:

Create a new Kernel class.

self.py_kernels should be a map from DispatchKey to Kernel.
Kernel stores a flag on if the keyset is necessary. Kernel also has a `__call__` on it.

Before invoking `kernel(*args, **kwargs)`, check kernel.with_keyset. If this is true, thread the keyset through.

Reply:
Lol @yanboliang this branch isn't being hit. This is HigherOrderOperator.dispatch